### PR TITLE
gaphor: 2.8.2 -> 2.26.0

### PR DESCRIPTION
**Pull request by alex-nt**
## Description of changes

`gaphor` no longer builds, this will update it to `2.26.0`.

## Things done

* Added [better-exceptions](https://pypi.org/project/better-exceptions/) as it was the only missing package needed in latest `gaphor`
* Added all the new dependencies `gaphor` has

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/better-exceptions/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  build-system = [ setuptools ];
+
+  pythonImportsCheck = [ "better_exceptions" ];
+
+  # As noted by @WolfangAukang, some check files need to be disabled because of various errors, same with some tests.
+  # After disabling and running the build, no tests are collected.
+  doCheck = false;
+
+  meta = {
+    description = "Pretty and more helpful exceptions in Python, automatically.";
```
Comment (dotlambda) [inline pkgs/development/python-modules/better-exceptions/default.nix:28]:
```suggestion
    description = "Pretty and more helpful exceptions in Python, automatically";
```
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

File: pkgs/tools/misc/gaphor/default.nix [hunk 2]
```diff
@@ -79,6 +96,6 @@ buildPythonApplication rec {
     maintainers = [ ];
     homepage = "https://github.com/gaphor/gaphor";
     license = licenses.asl20;
-    platforms = [ "x86_64-linux" ];
+    platforms = [ "x86_64-linux" "aarch64-linux" ];
```
Comment (dotlambda) [inline pkgs/tools/misc/gaphor/default.nix:99]:
```suggestion
    platforms = lib.platforms.linux;
```

File: pkgs/tools/misc/gaphor/default.nix [hunk 3]
```diff
@@ -79,6 +96,6 @@ buildPythonApplication rec {
     maintainers = [ ];
```
Comment (dotlambda) [inline pkgs/tools/misc/gaphor/default.nix:96]: Wanna add yourself?

Comment (alex-nt) [reply pkgs/tools/misc/gaphor/default.nix:96]: No for now, as I can't guarantee that I will keep it updated.

**Comment by natsukium** [comment]
@ofborg eval

File: pkgs/tools/misc/gaphor/default.nix [hunk 4]
```diff
@@ -4,51 +4,68 @@
...
-    hash = "sha256-+qqsSLjdY2I19fxdfkOEQ9DhTTHccUDll4O5yqtLiz0=";
+    hash = "sha256-e0K5bfgPqlJh8qrAz40c/w3ANzkfa/6txuqzQDJYXfE=";
   };
 
+  pythonRelaxDeps = [ "defusedxml" ];
+
   nativeBuildInputs = [
     copyDesktopItems
     gobject-introspection
     poetry-core
```
Comment (natsukium) [inline pkgs/tools/misc/gaphor/default.nix:44]:
```suggestion
```

**Review by dotlambda — COMMENTED** [review]
The commit message should be `python312Packages.better-exceptions: init at 0.3.3`.
Please also bring the commits in a sensible order. The package set should evaluate after every commit.

**Review by leona-ya — APPROVED** [review]

**Pull request merged**